### ISSUE TITLE
Lifetime of PyScaffold 3.* branch

### ISSUE THREAD

**Issue by FlorianWilhelm** [pyscaffold/pyscaffold#537]
Since there is already PyScaffold 4.1, we should think about when we officially slate PyScaffold 3.* as no longer supported. Maybe even we should even be removing the part in the docs mentioning PyScaffold 2.* for Python 2 projects as hopefully no one is starting new Python 2 projects anymore.

**Comment by abravalheri** [comment]
Historically we have been maintaining the latest major version behind as kind of *courtesy* "LTS" policy, right?

What makes me think about v5... the initial plan was to wait for PEP 621 to land on `setuptools`, which I though would happen in the middle of this year. But I really overestimated the willingness of the community to contribute to that project (or underestimated the complexity of the code 😝).

If we want to keep the same *courtesy* "LTS" policy, but drop the PEP 621 requirement for v5, that could be planned...

**Comment by FlorianWilhelm** [comment]
> Historically we have been maintaining the latest major version behind as kind of courtesy "LTS" policy, right?

That's right, I just wonder (and am completely open about all directions) if we should have some kind of rule of thumb. This could be

1.  (status quo): The latest two major version branches are supported.
2. (time based): After a major version is superseded by the next major version, there is a transition period of N years.
3. (minor version): With the first increment of the minor version the previous major version is deprecated. So with 4.1 we would deprecate the 3.* branch.

My gut feeling says that 2. could be a viable option.

**Comment by abravalheri** [comment]
How about whatever is happens first between 1 and 2, with N=1 year? 😝

**Comment by FlorianWilhelm** [comment]
> How about whatever happens first between 1 and 2, with N=1 year? 😝

Sounds good!

**Comment by abravalheri** [comment]
Just a minor comment: this should not affect the users too much because of the following aspects:

- PyScaffold usage is mostly "one shot"
- It's not a runtime dependency
- We currently provide either automatic "update" or at least instructions for manual update
